tests/end-to-end/upgrade/main.go

```diff
@@ -127,6 +133,15 @@ func main() {
 	dexConfig, err := getDexConfigFromCluster(k8sCli, cfg.DexUserSecret, cfg.DexNamespace, domainName)
 	fatalOnError(err, "while reading dex config from cluster")
 
+	servingCli, err := serving.NewForConfig(k8sConfig)
+	fatalOnError(err, "while reading serving config from cluster")
```

Comment:
The message is not aligned with what you're actually doing -> generating a client